### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by rsnazario**
Follow Map of the Testing 
 
  - Added Rspec files
  - Added structure of rspec testing for the methods
  - Added comparison to the main tests of the ruby-documentation
  - Applied rubocop checking

### CODE REVIEW

File: spec/enum_methods_spec.rb [hunk 1]
```diff
@@ -0,0 +1,128 @@
+# rubocop:disable Layout/LineLength
+
+require './enum_methods'
+
+RSpec.describe Enumerable do
+  let(:arr_1) { [1, 2, 3, 4, 5] }
+  let(:arr_2) { [1, 2, nil, 4] }
+  let(:arr_3) { ['cat', 3, true, 'board'] }
```
Comment (mosesogwo) [inline spec/enum_methods_spec.rb]: arr_3 is not used anywhere in the test. It should be removed.

File: spec/enum_methods_spec.rb [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+  describe '#my_all?' do
+    context 'if no block given return true if none of object values is nil or false' do
+      it { expect(arr_1.my_all?).to eq(true) }
+      it { expect([].my_all?).to eq(true) }
+    end
+    context 'if no block given return false if none one the object values is nil or false' do
+      it { expect(arr_2.my_all?).to eq(false) }
+    end
+    context 'if argument is Regular expression return true if all objects contain it' do
```
Comment (mosesogwo) [inline spec/enum_methods_spec.rb:55]: Add test for the converse of this context - when not all the objects match the regex

File: spec/enum_methods_spec.rb [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+      it { expect(%w[ant beat cat].my_all?(/t/)).to eq(true) }
+    end
+    context 'if argument is Class return true if all objects are of that class' do
```
Comment (mosesogwo) [inline spec/enum_methods_spec.rb:61]: Add test for the converse of this context - when not all the objects are of that class

File: spec/enum_methods_spec.rb [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+      it { expect(arr_1.my_all?(Integer)).to eq(true) }
+    end
+  end
+
+  describe '#my_any?' do
+    context 'if no block given return true if atleast one of object values is not nil or false' do
+      it { expect(arr_2.my_any?).to eq(true) }
+    end
+    context 'if argument is Regular expression return true if any objects contain it' do
```
Comment (mosesogwo) [inline spec/enum_methods_spec.rb:73]: Add test for the converse of this context.

File: spec/enum_methods_spec.rb [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+      it { expect(%w[dog bear cat].my_any?(/t/)).to eq(true) }
+    end
+    context 'if argument is Class return true if any object is of that class' do
```
Comment (mosesogwo) [inline spec/enum_methods_spec.rb:79]: Add test for the converse of this context.

File: spec/enum_methods_spec.rb [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+    context 'if no block given and object is empty return false' do
+      it { expect([].my_any?).to eq(false) }
+    end
+  end
+
+  describe '#my_none?' do
+    context 'if no block given return true only if none of collection members is true' do
+      it { expect([nil, false].my_none?).to eq(true) }
+    end
+    context 'if argument is Regular expression return false if all any value of object contains it' do
```
Comment (mosesogwo) [inline spec/enum_methods_spec.rb]: Add test for the converse of this context.

File: spec/enum_methods_spec.rb [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+      it { expect(%w[ant beat cat].my_none?(/t/)).to eq(false) }
+    end
+    context 'if argument is Class return true if all any value of object is not of that class' do
```
Comment (mosesogwo) [inline spec/enum_methods_spec.rb]: Add test for the converse of this context.

File: spec/enum_methods_spec.rb [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+      it { expect(arr_1.my_count).to eq(arr_1.length) }
+    end
+    context 'return the number of even numbers on the array' do
+      it { expect(arr_1.my_count(&:even?)).to eq(arr_1.count(&:even?)) }
+    end
+  end
+
+  describe '#my_map' do
+    context 'if no block given return enumerator' do
+      it { expect(arr_1.my_each_with_index).to be_a(Enumerator) }
```
Comment (mosesogwo) [inline spec/enum_methods_spec.rb]: You're testing for my_each_with_index instead of my_map

**Review by mosesogwo — CHANGES_REQUESTED** [review]
**Hello, good work with testing this project :+1: See my comments for the required changes.**

Also, add test for your multiply_els method.


_**Kindly make the changes and submit another code review request.**_

**Any question? You can reach me on Slack (Moses Ogwo)**

**Cheers:stars:**

### Status: Changes Requested.

File: spec/enum_methods_spec.rb [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+      it { expect(arr_1.my_each).to be_a(Enumerator) }
+    end
+  end
+
+  describe '#my_each_with_index' do
+    context 'if no block given return enumerator' do
+      it { expect(arr_1.my_each_with_index).to be_a(Enumerator) }
+    end
+    context 'return object after application of block given requirements' do
+      hash = {}
```
Comment (euqueme) [inline spec/enum_methods_spec.rb:25]: [OPTIONAL]
- [ ] Define the hash variable on line #25 also like a let variable within the '#my_each_with_index' block

**Review by euqueme — APPROVED** [review]
> ### Pull Request under the review by Maria Eugenia Queme P. (TSE)

### Hello @rsnazario and @Elbie-em :balloon: :lollipop: 

### Thank you for doing the changes requested by the previous reviewer I can see you did a good amount of research, :medal_sports:  You've added a good amount of test cases so your project is good to pass  :heavy_check_mark: 

[OPTIONAL]

- [ ] I won't stop you from moving forward for this but in the test cases you have for example

```ruby
expect (value).to eq(false)
```
you could also validate like this:
```ruby
expect(value).not_to eq(true)
```
### Please :pray: do the optional changes (or not) :man_technologist: :man_technologist: and merge this branch always checking all your test cases pass :memo: 

Happy coding! :rocket: 

Any question? ask me on slack **@ Maru**

PROJECT STATUS: APPROVED :green_apple:

**Pull request merged**